### DISCUSSION TITLE
Automation

### DISCUSSION THREAD

**Discussion by mixmonkey** [apohl79/audiogridder#433] [Ideas]
Many thanks for adding the ability to join plugin parameters to the host automation! 

I use Reaper with the Control Surface Integrator extension and now I can control the plugin parameters on the remote machine from my control surface (Mackie C4) Fantastic!:)

One small feature request, could you make a button that would assign all the plugin parameters to AudioGridder with one push- the equivalent of ticking every parameter shown in the "Automation" menu option, so that plugin parameter 0 maps to AudioGridder parameter 0, 1 to 1 etc.

I increased the number of available 'slots' from 16 to 1024 and assigned all the DIVA soft synth parameters manually. It took a while, but the map I use with CSI and the C4 locally works fine via AudioGridder on the remote machine as well. 

The only improvement I could think of would be to pass through the parameter feedback from the remote plugin unchanged. At the moment it gets changed to the default 0.0 -1.0 from a more useful feedback, frequency in Hz for instance. 

Anyway, truly great work. Many thanks and donation sent.

**Comment by apohl79** [comment]
> I use Reaper with the Control Surface Integrator extension and now I can control the plugin parameters on the remote machine from my control surface (Mackie C4) Fantastic!:)

nice!

> One small feature request, could you make a button that would assign all the plugin parameters to AudioGridder with one push- the equivalent of ticking every parameter shown in the "Automation" menu option, so that plugin parameter 0 maps to AudioGridder parameter 0, 1 to 1 etc.

Thats a great idea, will add a feature request ticket for this!

> The only improvement I could think of would be to pass through the parameter feedback from the remote plugin unchanged. At the moment it gets changed to the default 0.0 -1.0 from a more useful feedback, frequency in Hz for instance.

This is not possible with the current plugin frameworks. I tried this when writing the automation code. Unfortunately the hosts wont let a plugin change the parameter type, range etc later on after initialization, so everything is just 0-1.

> Anyway, truly great work. Many thanks and donation sent.

Thanks a lot!

**Comment by mixmonkey** [comment]
> This is not possible with the current plugin frameworks. I tried this when writing the automation code. Unfortunately the hosts wont let a plugin change the parameter type, range etc later on after initialization, so everything is just 0-1.

Ah, I see. Not a problem. Acustica plugins suffer from the same drawback (the format of the feedback being fixed after initialization, so when they offer different EQ types within the same plugin, say, they can't change the feedback to match). 0-1 is fine:)

I'll definitely be scaling up my use of AudioGridder now I can map remote plugins to my control surface.
